CHANGELOG.md

```diff
@@ -0,0 +1,13 @@
+## 2024-02-12
+
+### Changed
+- Added --build-windows and --build-check flags to build scripts to allow optional skipping of Windows build and unit tests/mypy checks.
+
+### Fixed
+- Resolved issues with paho-mqtt upgrade breaking cloudadapter's mqtt connections.
+- Improved error messaging for failed URL fetches.
+- Addressed missing GUID when not provided by manifest for fwupdate tool.
+- Fixed SOTA snapshot conditions to prevent double reboots on EXT4 systems.
+
+### Security
+- Updated cryptography to address CVE-2023-5678 and CVE-2023-6129 vulnerabilities.
\ No newline at end of file
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Maintainability$</h3>

Consider adding a newline at the end of the CHANGELOG.md to adhere to POSIX standards and ensure compatibility with Unix-like systems. This minor change can improve file processing in various environments.